### ISSUE TITLE
Client Side Uploading via Ajax 

### ISSUE THREAD

**Issue by denismars** [balderdashy/skipper#69]
Here's a simple example code to hopefully save you time trying to figure out how to to get ajax uploader to work with Skipper.

```

    <form id="uploadForm">
        <input type="text" name="foo" placeholder="some other metadata"/>
        <input type="file" name="nameHere"/>
    </form>

    var formData = new FormData($('#uploadForm'));

    $.ajax({
        type : 'post',
        url:'/file/upload',
        data : formData,
        contentType : false,
        processData : false
    }).success(function(result) {
        console.log('Success uploading to Sails app', result);
    }).error(function(err) {
        console.log('Error uploading file', err);
    });

```
The key is that skipper supports Multipart/Form-Data so to get your ajax to work we need to make the ajax call formatted to transmit it's data as a FormData such that req.file('nameHere') can pick it up and process it correctly.

**Comment by martingg88** [comment]
yes. this will exclude empty field value. any idea how to include empty field value as well when sending over to server.

**Comment by cburatto** [comment]
How about serializing the form and sending as JSON? Might not be what you need, but I'd slowly try to move towards a more pure JSON interaction, leaving form data behind. Treat everything as an API call...

**Comment by martingg88** [comment]
checked with this https://github.com/balderdashy/skipper/pull/61/commits. 
They are trying to include empty value field.
But the file is not committed yet.

**Comment by martingg88** [comment]
checked with this https://github.com/balderdashy/skipper/pull/61/commits. 
They are trying to include empty value field.
But the file is not committed yet.

**Comment by sailsbot** [comment]
Thanks for posting, @denismars.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
